### ISSUE TITLE
Question about Figure 4 and Section 4.1.3

### ISSUE THREAD

**Issue by FlorianHuo** [step-law/steplaw#6]
Thank you for your great work!

I have a question about Figure 4. The caption states that blue contours represent the conventional decay schedule, while the red contours represent the fixed learning rate schedule. However, the legend in appears to show the opposite convention (red line: min_lr = peak_lr / 10; blue line: keep min_lr = 1e-5). Could you help clarify this discrepancy? Am I misunderstanding the figure?

<img width="522" alt="Image" src="https://github.com/user-attachments/assets/12fd8dbe-6ed8-40df-b836-e1b187a37a96" />

Additionally, regarding your claim that the conventional decay schedule's learning rate and batch size are suboptimal—was this conclusion drawn by comparing the shape of the contours, or did you also compare the loss between the two schedules? 

<img width="474" alt="Image" src="https://github.com/user-attachments/assets/fb713cfe-dc86-47c6-80af-4d9627a1255e" />

Thanks again for your time and for sharing your work! I look forward to your response.

**Comment by BillZid** [comment]
Thank you for your careful observation and thoughtful questions!

Regarding Figure 4: You’re absolutely right—there was an inconsistency between the legend and the caption in the original version. The caption text (“blue = conventional decay, red = fixed min_lr”) was correct, but the legend colors were accidentally swapped. We’ve fixed this in an updated version and appreciate you catching the mistake!

About the suboptimality of the conventional decay schedule:

Yes, we observed that a lower final min_lr (e.g., keeping it at 1e-5) generally leads to better loss, which aligns with empirical trends in SOTA training practices (e.g., many high-performance models avoid aggressive LR decay).
More importantly, our key point was that the conventional decay schedule unfairly penalizes larger peak LRs during the search process, which can distort the power-law (LR-batch size) relationship. The fixed min_lr approach mitigates this bias.
We appreciate your engagement with the work—let us know if you’d like further clarification!

**Comment by FlorianHuo** [comment]
Thank you for your detailed explanation, and I apologize for the delayed response.  

I’m particularly intrigued by the observation that a fixed minimal learning rate schedule outperforms the conventional decay schedule. This finding seems non-trivial, as maintaining a fixed minimal learning rate is often viewed as more aggressive—especially when considering the difference between the peak and final learning rates.

For instance, in your experiments, the peak learning rate ranges from 0.002 to 0.008. If decayed conventionally to 1/20 of the peak, the final learning rate would range from 1e-4 to 4e-4—10 to 40 times larger than the fixed minimal rate of 1e-5. This suggests that the fixed minimal schedule reduces the learning rate far more aggressively than conventional decay. 

Additionally, for the same peak learning rate, the fixed minimal schedule consistently results in a lower learning rate post-warmup compared to conventional decay. This could explain why conventional decay penalizes larger peak learning rates, as it effectively uses higher learning rates for most of the training. This might contribute to the left-skewed loss landscape observed.

To better understand this, I believe it would be insightful to compare the terminal losses of both schedules—or at least the two loss minima shown in the figure below. I’ve reviewed your open-sourced data but couldn’t locate experiments under the conventional decay setting. Did I overlook them, or have they not been released yet?

Thank you again for your clarification and for generously sharing your resources! I greatly appreciate your work and insights!

![Image](https://github.com/user-attachments/assets/fe6fc6e6-4ad0-4aba-8e9b-fe3438976096)
